### DISCUSSION TITLE
Buffering or Local Store and Forward of messages

### DISCUSSION THREAD

**Discussion by m2b** [thomassorensen2650/node-red-contrib-mqtt-sparkplug-plus#46] [Ideas]
In LTE gateways is not uncommon to lose connection to the cloud.   It would be nice if the system queues the messages when connection is lost and sends them once restored.   Queue can be in memory or disk (even better but not always supported in host).

**Comment by thomassorensen2650** [comment]
This functionality has been there for a while, but it never really worked that well.. I just merged a PR that will add more robust support for store-forward/buffering. It will also supports disk backed store-forward.

Should be released in a few weeks.

**Comment by thomassorensen2650** [comment]
This is implemented in version 2.1.0
